core/src/main/scala/org/http4s/parser/HttpHeaderParser.scala

```diff
@@ -72,7 +73,16 @@ object HttpHeaderParser extends SimpleHeaders
   def parseHeader(header: Header.Raw): ParseResult[Header] = {
     allParsers.get(header.name) match {
       case null => ParseResult.success(header) // if we don't have a rule for the header we leave it unparsed
-      case parser => parser(header.value)
+      case parser =>
+        try parser(header.value)
+        catch {
+          // We need a way to bail on invalid dates without throwing.  There should be a better way.
+          case e: ParseFailure =>
+            ParseResult.success(header)
+          case e: InvocationTargetException if e.getCause.isInstanceOf[ParseFailure] =>
```

Comment:
I am really not proud of this section.  We are throwing an exception out of the date parser when it's well formed, but not a date.  And then because of the reflection in the parser, it can be wrapped.

I think we should ban throwing exceptions from parsers altogether, but I don't know how to express this in parboiled2.